### PR TITLE
doc: improve Drain examples

### PR DESCRIPTION
**Pull request by tshepang**
Second sentence actually repeats info from first sentence. "from start to end" also feels like it adds nothing.

I also extended Vec::drain example.

### CODE REVIEW

File: src/libcollections/string.rs [hunk 1]
```diff
@@ -1015,8 +1015,7 @@ impl String {
     }
 
     /// Create a draining iterator that removes the specified range in the string
-    /// and yields the removed chars from start to end. The element range is
-    /// removed even if the iterator is not consumed until the end.
```
Comment (bluss) [inline src/libcollections/string.rs]: The information about removal is good I think. You might otherwise think that only elements extracted from the iterator are removed from the String or Vec.

Comment (tshepang) [inline src/libcollections/string.rs]: @bluss it's only the extracted elements that are removed

Comment (bluss) [inline src/libcollections/string.rs]: No it's not, I implemented this for Vec and String.

Comment (tshepang) [inline src/libcollections/string.rs]: Not sure I understand you. Following does not panic:

    let mut v = vec![1, 2, 3];
    assert_eq!(vec![3], v.drain(2..).collect::<Vec<_>>());
    assert_eq!(vec![1, 2], v);
    v.drain(..);
    assert!(v.is_empty());

Comment (bluss) [inline src/libcollections/string.rs]: ok, I mean that:

```rust
let mut v = vec![1, 2, 3];
v.drain(..).take(1).collect::<Vec<_>>();
```

All elements in the range are removed even if  only 1 element is taken ("extracted") from the iterator.

Comment (tshepang) [inline src/libcollections/string.rs]: Now I understand. But that is somewhat orthogonal isn't it... not worth mentioning in the summary. The examples I added also demonstrate that fact.

Comment (bluss) [inline src/libcollections/string.rs]: It should be mentioned somewhere in the docs for the method IMO. It's something users have already wondered about, always best to say it explicitly.

Comment (tshepang) [inline src/libcollections/string.rs]: alright

Comment (tshepang) [inline src/libcollections/string.rs]: I re-instated the info @bluss

**Comment by bluss** [comment]
Thanks!

@bors r+
